### PR TITLE
Origin/feature/sm 4697 display task list in report details screen

### PR DESCRIPTION
**Pull request by meet-saferme**
Added the code related to fetching the task, team users and update the few already available classes

### CODE REVIEW

File: app/src/main/kotlin/com/thundermaps/apilib/android/api/resources/TeamUsersResource.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.thundermaps.apilib.android.api.resources
+
+import com.thundermaps.apilib.android.api.requests.RequestParameters
+import com.thundermaps.apilib.android.api.responses.models.Result
+import com.thundermaps.apilib.android.api.responses.models.TeamUserDetails
+
+interface TeamUsersResource {
```
Comment (liemvo) [inline app/src/main/kotlin/com/thundermaps/apilib/android/api/resources/TeamUsersResource.kt]: Team Resource can provide the TeamUser. We can update TeamResource to get the list of TeamUser

File: app/src/main/kotlin/com/thundermaps/apilib/android/api/responses/models/TeamUserDetails.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package com.thundermaps.apilib.android.api.responses.models
+
+import com.google.gson.annotations.Expose
+import com.google.gson.annotations.SerializedName
+
+class TeamUserDetails(
```
Comment (liemvo) [inline app/src/main/kotlin/com/thundermaps/apilib/android/api/responses/models/TeamUserDetails.kt]: Name should be TeamUser

File: app/src/main/kotlin/com/thundermaps/apilib/android/impl/SaferMeClientImpl.kt [hunk 3]
```diff
@@ -43,7 +45,8 @@ class SaferMeClientImpl @Inject constructor(
     private val brandResourceImpl: BrandResourceImpl,
     private val shapeResourceImpl: ShapeResourceImpl,
     private val categoryResourceImpl: CategoryResourceImpl,
-    private val formResourceImpl: FormResourceImpl
+    private val formResourceImpl: FormResourceImpl,
+    private val teamUserResourceImpl: TeamUsersResourceImpl
```
Comment (liemvo) [inline app/src/main/kotlin/com/thundermaps/apilib/android/impl/SaferMeClientImpl.kt]: can remove it

File: app/src/main/kotlin/com/thundermaps/apilib/android/impl/resources/TeamUsersResourceImpl.kt [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import io.ktor.client.request.url
+import io.ktor.http.HttpMethod
+import io.ktor.util.KtorExperimentalAPI
+import java.net.UnknownHostException
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@KtorExperimentalAPI
+@Singleton
+class TeamUsersResourceImpl @Inject constructor(
```
Comment (liemvo) [inline app/src/main/kotlin/com/thundermaps/apilib/android/impl/resources/TeamUsersResourceImpl.kt]: Missing test

**Review by liemvo — CHANGES_REQUESTED** [review]

**Comment by liemvo** [comment]
The branch is out of date

File: app/src/main/kotlin/com/thundermaps/apilib/android/api/resources/TeamUsersResource.kt (hunk 1 above)
Comment (meet-saferme) [reply app/src/main/kotlin/com/thundermaps/apilib/android/api/resources/TeamUsersResource.kt]: Ok will add it into the TeamResource

File: app/src/main/kotlin/com/thundermaps/apilib/android/api/responses/models/TeamUserDetails.kt (hunk 2 above)
Comment (meet-saferme) [reply app/src/main/kotlin/com/thundermaps/apilib/android/api/responses/models/TeamUserDetails.kt]: WIll rename it

File: app/src/main/kotlin/com/thundermaps/apilib/android/impl/SaferMeClientImpl.kt (hunk 3 above)
Comment (meet-saferme) [reply app/src/main/kotlin/com/thundermaps/apilib/android/impl/SaferMeClientImpl.kt]: Will remove TeamUsersResourceImpl

File: app/src/main/kotlin/com/thundermaps/apilib/android/impl/resources/TeamResourceImpl.kt [hunk 5]
```diff
@@ -46,7 +48,37 @@ class TeamResourceImpl @Inject constructor(
...
+        }
+        val call = getTeamUsersCall(parameters, teamId)
+        return resultHandler.processResult(call, gson)
+    }
+
     companion object {
+        private const val TEAM_USER_PATH = "team_users"
+        @VisibleForTesting
+        const val FIELDS_PARAM =
+            "fields=first_name,last_name,email"
```
Comment (liemvo) [inline app/src/main/kotlin/com/thundermaps/apilib/android/impl/resources/TeamResourceImpl.kt]: one line

File: app/src/main/kotlin/com/thundermaps/apilib/android/impl/resources/TeamResourceImpl.kt [hunk 6]
```diff
@@ -46,7 +48,37 @@ class TeamResourceImpl @Inject constructor(
...
+        val call = client.call(HttpRequestBuilder().takeFrom(requestBuilder).apply {
+            method = HttpMethod.Get
+            url(AndroidClient.baseUrlBuilder(parameters).apply {
+                encodedPath = "$encodedPath$PATH/$teamId/$TEAM_USER_PATH?$FIELDS_PARAM"
+            }.build())
+        })
+        return call
+    }
+
+    override suspend fun getTeamUsers(
```
Comment (liemvo) [inline app/src/main/kotlin/com/thundermaps/apilib/android/impl/resources/TeamResourceImpl.kt:65]: Missing test

**Review by liemvo — CHANGES_REQUESTED** [review]
The branch is out of date

**Review by liemvo — APPROVED** [review]

**Pull request merged**